### PR TITLE
Queryview dataset decorator

### PR DESCRIPTION
**Pull request by IvoTod**

### CODE REVIEW

**Comment by IvoTod** [comment]
[SP27: Add decorator for getting dataset](https://trello.com/c/ZRmiOXVC/1276-sp27-add-decorator-for-getting-dataset)

File: wdae/wdae/query_base/decorators.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+from utils.datasets import find_dataset_id_in_request
+
+
+def inject_dataset(func):
+    """Injects a dataset property into the request of a view method"""
+    def wrapper(self, request, dataset, *args, **kwargs):
+        dataset_id = find_dataset_id_in_request()
+        if dataset_id is None:
```
Comment (nikolaystanishev) [inline wdae/wdae/query_base/decorators.py]: You should try to pass None id to the facade. Expected result should be None.

File: wdae/wdae/datasets_api/permissions.py [hunk 2]
```diff
@@ -2,28 +2,25 @@
...
-    def has_permission(self, request, view):
-        dataset_id = request.query_params.get('dataset_id', None)
-        if dataset_id is None:
-            dataset_id = request.query_params.get('datasetId', None)
-        if dataset_id is None:
-            dataset_id = request.data.get('datasetId', None)
-        if dataset_id is None:
-            dataset_id = request.parser_context. \
-                get('kwargs', {}).get('common_report_id', None)
+    @staticmethod
```
Comment (nikolaystanishev) [inline wdae/wdae/datasets_api/permissions.py]: This method is overriding method from base class which is not staticmethod.

File: wdae/wdae/datasets_api/permissions.py [hunk 3]
```diff
@@ -2,28 +2,25 @@
...
         if dataset_id is None:
             return True
 
-        return self.has_object_permission(request, view, dataset_id)
+        return IsDatasetAllowed.has_object_permission(
+                request, view, dataset_id)
 
-    def has_object_permission(self, request, view, dataset_id):
-        return self.user_has_permission(request.user, dataset_id)
+    @staticmethod
```
Comment (nikolaystanishev) [inline wdae/wdae/datasets_api/permissions.py]: This method is overriding method from base class which is not staticmethod.

File: wdae/wdae/query_base/decorators.py [hunk 4]
```diff
@@ -0,0 +1,13 @@
+from utils.datasets import find_dataset_id_in_request
+
+
+def inject_dataset(func):
```
Comment (nikolaystanishev) [inline wdae/wdae/query_base/decorators.py]: Try using this decorator in the datasets_api view

**Pull request merged**